README.md

```diff
@@ -48,6 +48,49 @@ var consulCreds = await vaultClient.V1.Secrets.Consul.GetCredentialsAsync(consul
 var consulToken = consulCredentials.Data.Token;
 ```
 
+### How to reuse Vault Client Token?
+
+* **TLDR:** The same Vault Client Token will be intelligently **reused** for the lifetime of the client.
+* **NOTE:** The Vault Client Token is lazy loaded; it becomes available via the ReturnedLoginAuthInfo method once the first Vault API is requested.
+Example:
+```cs
+ // Setup Vault Client via Github Auth Method
+ IAuthMethodInfo authMethod = new GitHubAuthMethodInfo(personalAccessToken);
+ var vaultClientSettings = new VaultClientSettings("https://MY_VAULT_SERVER:8200", authMethod);
+ IVaultClient vaultClient = new VaultClient(vaultClientSettings);
+ 
+ // If the token is attempted to be read at this time; it is null as it is lazy loaded on first Vault API call
+ // var token = authMethod.ReturnedLoginAuthInfo.ClientToken; -->null!
+
+ // Use client to read a key-value secret and trigger the load of ClientToken
+var kv2Secret = await vaultClient.V1.Secrets.KeyValue.V2.ReadSecretAsync("secret-name");
+ // now ReturnedLoginAuthInfo method will provide a valid ClientToken for further use
+ var token = authMethod.ReturnedLoginAuthInfo.ClientToken;
+```
+* The Vault Client Token can be persisted and shared across clients via a store like Redis. Redis TTL capabilities can be leveraged so that the token will be expired automatically using the returned LeaseDurationSeconds value.
```

Comment:
@leowmjw I am okay with the first part of the doc update, which explains lazy loading.

Would prefer to remove the second part, which talks about sharing of the token via Redis etc. In general, the vault token is highly confidential, and caches are not the best places for passwords. It is the host application's prerogative as to how they want to deal with secrets, as per their security requirements. I don't think we should prescribe (or even mislead) folks in this library's docs.

Let me know thoughts.